framework/test/mocha/common/utils/blocks_and_transactions_helper.js

```diff
@@ -0,0 +1,250 @@
+		recipientId: random.account().address,
+		passphrase: account.passphrase,
+	});
+}
+
+function createCreditTransaction(account, amount) {
+	return transfer({
+		amount: new Bignum(NORMALIZER).multipliedBy(amount).toString(),
+		recipientId: account.address,
+		passphrase: accountFixtures.genesis.passphrase,
+	});
+}
+
+const formatTransaction = t => ({
+	id: t.id,
+	amount: new Bignum(t.amount).toFixed(),
+	senderId: t.senderId,
+	recipientId: t.recipientId,
+});
+
+/**
+ * Blocks and Transactions Helper
+ */
+class BAT {
+	constructor(library) {
+		this._library = library;
+		this._transactions = [];
+
+		this.promisifyProcessBlock = util.promisify(
+			this._library.modules.blocks.verify.processBlock
+		);
+
+		this.txPool = this._library.rewiredModules.transactions.__get__(
+			'__private.transactionPool'
+		);
+	}
+
+	async initAccountAndCredit({ amount, account }) {
+		this._account = account || random.account();
+		const transaction = createCreditTransaction(this._account, amount);
+		await addTransactionsAndForge(this._library, [transaction], 0);
+	}
+
+	getAccount() {
+		return this._account;
+	}
+
+	add(amount, type, expected, sender) {
+		const transaction = {
+			type,
+			amount,
+			expect: expected,
+			data: type(sender || this._account, amount),
+		};
+		this._transactions.push(transaction);
+	}
+
+	getAllTransactions() {
+		return this._transactions.map(t => t.data);
+	}
+
+	cleanTransactions() {
+		this._transactions = [];
+	}
+
+	getTransactionsInLastBlock() {
+		const lastBlock = this._library.modules.blocks.lastBlock.get();
+
+		// We return only transaction ID, amount (in string format), sender and recipient
+		return lastBlock.transactions.map(formatTransaction);
+	}
+
+	getValidSortedTransactions() {
+		const validTransactions = this._transactions
+			// Get only transactions marked as valid
+			.filter(t => t.expect === EXPECT.OK)
+			// Amounts have to be instances of Bignum for sorting
+			.map(t => initTransaction.fromJson(t.data));
+
+		// Sort transactions the same way as they are sorted in a block
+		const sortedTransactions = BlockLogic.sortTransactions(validTransactions);
+
+		// We return only transaction ID, amount (in string format), sender and recipient
+		return sortedTransactions.map(formatTransaction);
+	}
+
+	isValidTransactionInPool() {
+		return (
+			this._transactions
+				.filter(t => t.expect === EXPECT.OK)
+				.filter(t => isTransactionInPool(this._library, t.data.id)).length > 0
+		);
+	}
+
+	getTransactionsInPool() {
+		// Get only transactions that exists in transaction pool
+		const transactionsInPool = this._transactions
+			.filter(t => isTransactionInPool(this._library, t.data.id))
+			.map(t => t.data);
+
+		// We return only transaction ID, amount (in string format), sender and recipient
+		return transactionsInPool.map(formatTransaction);
+	}
+
+	async enqueueTransactions() {
+		return Promise.mapSeries(this._transactions, t =>
```

Comment:
prefer
```
for (const t of this._transactions) {
  await addTransaction(this._library, t.data);
}
```

Reply:
`no-restricted-syntax`, `no-await-in-loop`, if we finally want to allow this - change the rules first.

Reply:
I'd just avoid using bluebird methods.